python/ray/_raylet.pyx

```diff
@@ -2732,6 +2737,48 @@ cdef void terminate_asyncio_thread() nogil:
         core_worker = ray._private.worker.global_worker.core_worker
         core_worker.stop_and_join_asyncio_threads_if_exist()
 
+
+cdef void call_actor_shutdown() noexcept nogil:
+    """C++ wrapper function that calls the Python actor shutdown callback."""
+    with gil:
+        _call_actor_shutdown()
+
+
+def _call_actor_shutdown():
+    """Internal function that calls actor's __ray_shutdown__ method."""
+    try:
+        worker = ray._private.worker.global_worker
+
+        if not worker.actors:
+            return
+
+        actor_id, actor_instance = next(iter(worker.actors.items()))
+        if actor_instance is not None:
+            try:
+                actor_instance.__ray_shutdown__()
+            except Exception:
+                logger.exception("Error during actor __ray_shutdown__ method")
+            finally:
+                worker.actors.pop(actor_id, None)
```

Comment:
you're sure we don't depend on `worker.actors` anywhere after this executes, right?